src/core_init_atmosphere/mpas_stack.F

```diff
@@ -0,0 +1,273 @@
+module mpas_stack
+    
+   implicit none
+
+   private
+
+   ! Public Subroutines and Structures
+   public :: mpas_stack_is_empty
+   public :: mpas_stack_push
+   public :: mpas_stack_pop
+   public :: mpas_stack_free
+
+   public :: node, payload_t
+
+   type payload_t
+   end type payload_t
+
+   type node
+      type (node), pointer :: next => null()
+      class (payload_t), pointer :: payload => null()
+   end type node
```

Comment:
Since this type is made public by the module, we could consider naming it with an `mpas_` prefix, and similarly for `payload_t`. Also, if the `_t` suffix is used for the payload type, perhaps we should use it for the node type as well, and consider using `_type`, which is used elsewhere in MPAS infrastructure? Also, one can imagine all sorts of situations involving "nodes" in MPAS infrastructure, so perhaps qualifying that this is a *stack* node would be good. In all, then, how about `mpas_stack_node_type` and `mpas_stack_payload_type`?

If `mpas_stack_node_type` is too long, and if the module supports pushing entire stacks onto each other, then `mpas_stack_type` might be appropriate. By allocating a variable of type `mpas_stack_type`, the user is effectively creating a stack with one item.

Reply:
Having looked more at how the stack works, I think it seems reasonable to name the stack derived type as `mpas_stack_type`. The user doesn't really need to know that the stack is implemented as a linked list -- only that they have a pointer to an `mpas_stack_type`, onto which they can push or pop payloads.

Reply:
That sounds good to me, and I agree, best to try to get unique as names as possible. I didn't even think about naming the stack and payload in such a way, so thanks for the recommendation.

So, just to confirm, I'll rename `node` to be: `mpas_stack_type` and I'll rename `payload_t` to be `mpas_stack_payload_type`?

Reply:
Yes, `mpas_stack_type` and `mpas_stack_payload_type` sound good to me.